contract_line_extended/models/account_analytic_invoice_line.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 Therp BV.
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+
+
+SQL_SYNCHRONIZE1 = \
+    """UPDATE account_analytic_invoice_line
+ SET contract_id = analytic_account_id
+ WHERE contract_id IS NULL"""
+
+SQL_SYNCHRONIZE2 = \
+    """UPDATE account_analytic_invoice_line
+SET partner_id=subquery.partner_id,
+    date_start=subquery.date_start,
+    date_end=subquery.date_end
+ FROM (SELECT l.id, c.partner_id, c.date_start, c.date_end
+      FROM account_analytic_invoice_line l
+      JOIN account_analytic_account c
+      ON l.contract_id = c.id
+    WHERE (l.partner_id is NULL AND NOT c.partner_id IS NULL)
+    OR (l.date_start is NULL AND NOT c.date_start IS NULL)
+    OR (l.date_end is NULL AND NOT c.date_end IS NULL))
+ AS subquery
+ WHERE subquery.id = account_analytic_invoice_line.id"""
+
+
+class AccountAnalyticInvoiceLine(models.Model):
+    _inherit = 'account.analytic.invoice.line'
+
+    # We already had analytic_account_id to refer to the contract to which
+    # this line belongs. Unfortunately this is overwritten by
+    # account.analytic.contract.line which (ab)uses the same field to refer
+    # to the contract template.
```

Comment:
Hmmm this feels to me like I broke something with the new implementation while being lazy. Am I understanding correctly that after turning a contract template (`account.analytic.contract`) into an actual contract (`account.analytic.account`), the `account.analytic.account.line`s that are created maintain a reference ID corresponding to the `account.analytic.contract.line` instead of being rewritten correctly? If so this, was not my intent and seems like an extension of #80